apps/prairielearn/src/models/course-instance.sql

```diff
@@ -0,0 +1,85 @@
+-- BLOCK check_course_instance_is_public
+SELECT
+  ci.share_source_publicly
+FROM
+  course_instances AS ci
+WHERE
+  ci.id = $course_instance_id;
+
+-- BLOCK select_assessments
+WITH
+  issue_count AS (
+    SELECT
+      a.id AS assessment_id,
+      count(*) AS open_issue_count
+    FROM
+      assessments AS a
+      JOIN issues AS i ON (i.assessment_id = a.id)
+    WHERE
+      a.course_instance_id = $course_instance_id
+      AND i.course_caused
+      AND i.open
+    GROUP BY
+      a.id
```

Comment:
query copied directly from instructorAssessments.sql so that it can be re-used both places.